src/external_account.rs

```diff
@@ -47,10 +48,33 @@ pub enum CredentialSource {
         /// file
         file: String,
     },
-    // TODO: Microsoft Azure and URL-sourced credentials
+    /// Microsoft Azure and URL-sourced credentials
+    Url {
+        /// url
+        url: String,
+        /// headers
+        headers: Option<HashMap<String, String>>,
+        /// format
+        format: UrlCredentialSourceFormat,
+    },
     // TODO: executable-sourced credentials
 }
 
+/// JSON schema of URL-sourced credentials' format.
+#[derive(Serialize, Deserialize, Debug, Clone)]
+#[serde(tag = "type")]
+pub enum UrlCredentialSourceFormat {
```

Comment:
This enum represents the format of document that the url returns. I avoid `UrlFormat` since it sounds "the format of url". I don't think `UrlCredentialSourceFormat` is good. Suggestion welcome.

Reply:
Using modules is another option 
```rust
pub mod credential_source {
    pub mod url {
        pub enum Format { ... }
    }
}
```